package.json

```diff
@@ -223,5 +223,10 @@
     "suiteNameTemplate": "{filepath}",
     "classNameTemplate": "{classname}",
     "titleTemplate": "{title}"
+  },
+  "eslintConfig": {
+    "rules": {
+      "no-shadow": 2
+    }
```

Comment:
We've already got `.eslintrc.json` so this configuration should be moved to that file:

```diff
diff --git a/.eslintrc.json b/.eslintrc.json
index 3e6205b8..15b3f9dd 100644
--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -4,6 +4,7 @@
     "import/order": [
       "error",
       { "newlines-between": "always-and-inside-groups" }
-    ]
+    ],
+    "no-shadow": 2
   }
 }
diff --git a/package.json b/package.json
index 25b5a262..4b81064b 100644
--- a/package.json
+++ b/package.json
@@ -223,10 +223,5 @@
     "suiteNameTemplate": "{filepath}",
     "classNameTemplate": "{classname}",
     "titleTemplate": "{title}"
-  },
-  "eslintConfig": {
-    "rules": {
-      "no-shadow": 2
-    }
   }
 }
```

With that change, running `npm run lint` now yields the following:

```
> @jeremyckahn/farmhand@1.15.25 lint
> eslint src --max-warnings=0


/farmhand/src/components/AchievementsView/AchievementsView.js
  29:29  error  'achievements' is already declared in the upper scope on line 11 column 8  no-shadow

/farmhand/src/components/AnimatedNumber/AnimatedNumber.js
   7:26  error  'number' is already declared in the upper scope on line 3 column 16  no-shadow
   9:27  error  'number' is already declared in the upper scope on line 3 column 16  no-shadow
  27:20  error  'number' is already declared in the upper scope on line 9 column 27  no-shadow

/farmhand/src/components/AppBar/AppBar.js
  34:27  error  'money' is already declared in the upper scope on line 14 column 25  no-shadow

/farmhand/src/components/CowCard/CowCard.js
  71:28  error  'id' is already declared in the upper scope on line 53 column 3  no-shadow

/farmhand/src/components/CowPen/CowPen.js
  132:20  error  'x' is already declared in the upper scope on line 76 column 42  no-shadow
  132:23  error  'y' is already declared in the upper scope on line 76 column 45  no-shadow

/farmhand/src/components/Farmhand/Farmhand.js
   714:34  error  'getPeerMetadata' is already declared in the upper scope on line 51 column 3  no-shadow
   799:15  error  'peerId' is already declared in the upper scope on line 797 column 16         no-shadow
  1201:18  error  'dialogView' is already declared in the upper scope on line 64 column 3       no-shadow

/farmhand/src/components/Field/Field.js
   59:3   error  'fieldMode' is already declared in the upper scope on line 19 column 10  no-shadow
  218:3   error  'fieldMode' is already declared in the upper scope on line 19 column 10  no-shadow
  238:14  error  '_null' is already declared in the upper scope on line 235 column 29     no-shadow
  309:5   error  'fieldMode' is already declared in the upper scope on line 19 column 10  no-shadow

/farmhand/src/components/LogView/LogView.js
  28:67  error  'i' is already declared in the upper scope on line 25 column 53  no-shadow

/farmhand/src/components/Navigation/Navigation.js
  316:16  error  'dialogView' is already declared in the upper scope on line 39 column 10  no-shadow

/farmhand/src/components/OnlinePeersView/OnlinePeersView.js
  32:8   error  'id' is already declared in the upper scope on line 25 column 3  no-shadow
  74:40  error  'id' is already declared in the upper scope on line 25 column 3  no-shadow

/farmhand/src/components/TierPurchase/TierPurchase.js
  44:11  error  'selectedTier' is already declared in the upper scope on line 25 column 10  no-shadow

/farmhand/src/components/Workshop/Workshop.test.js
  66:26  error  'gameState' is already declared in the upper scope on line 55 column 7  no-shadow

/farmhand/src/data/upgrades.js
  176:10  error  'toolType' is already declared in the upper scope on line 1 column 20  no-shadow

/farmhand/src/game-logic/reducers/changeCowAutomaticHugState.js
  25:36  error  'cow' is already declared in the upper scope on line 15 column 51  no-shadow

/farmhand/src/game-logic/reducers/generatePriceEvents.test.js
  50:15  error  'generatePriceEvents' is already declared in the upper scope on line 5 column 10  no-shadow

/farmhand/src/game-logic/reducers/harvestPlot.js
  82:46  error  'crop' is already declared in the upper scope on line 63 column 9  no-shadow

/farmhand/src/game-logic/reducers/makeRecipe.js
  18:6  error  'state' is already declared in the upper scope on line 12 column 28  no-shadow

/farmhand/src/game-logic/reducers/purchaseField.js
  21:10  error  '_' is already declared in the upper scope on line 19 column 33  no-shadow

/farmhand/src/game-logic/reducers/purchaseField.test.js
  30:15  error  'purchaseField' is already declared in the upper scope on line 3 column 10  no-shadow

/farmhand/src/game-logic/reducers/sellCow.test.js
  76:13  error  'cow' is already declared in the upper scope on line 10 column 7  no-shadow

/farmhand/src/game-logic/reducers/updateAchievements.js
  12:24  error  'state' is already declared in the upper scope on line 11 column 36  no-shadow

/farmhand/src/handlers/peer-events.js
   61:12  error  'id' is already declared in the upper scope on line 48 column 9   no-shadow
  169:12  error  'id' is already declared in the upper scope on line 164 column 9  no-shadow
  186:15  error  'id' is already declared in the upper scope on line 164 column 9  no-shadow

/farmhand/src/handlers/ui-events.js
  151:25  error  'fieldMode' is already declared in the upper scope on line 9 column 22   no-shadow
  181:7   error  'fieldMode' is already declared in the upper scope on line 9 column 22   no-shadow
  313:31  error  'dialogView' is already declared in the upper scope on line 9 column 10  no-shadow
  383:16  error  'e' is already declared in the upper scope on line 361 column 44         no-shadow

/farmhand/src/utils.js
  411:15  error  '_' is already declared in the upper scope on line 408 column 11                 no-shadow
  825:30  error  'farmProductsSold' is already declared in the upper scope on line 813 column 14  no-shadow

✖ 39 problems (39 errors, 0 warnings)
```

So, the real work of #388 is renaming the variable names that violate `no-shadow` to a sensible alternative.

Reply:
Done

Reply:
Thanks for moving the configuration to `.eslintrc.json`, @tarunsamanta2k20. However, the preexisting [`no-shadow` violations](https://github.com/jeremyckahn/farmhand/pull/389#discussion_r1090106907) remain. Those will all need to be fixed before this can be approved and merged.

Reply:
https://stackoverflow.com/questions/63961803/eslint-says-all-enums-in-typescript-app-are-already-declared-in-the-upper-scope is not related to the `no-shadow` violations that were reported. These violations are legitimate and need to be fixed before this can be merged. If those fixes aren't going to be a part of this PR, then it would be best to close it without merging.

Reply:
@jeremyckahn i am working on those fix